owid/catalog/datasets.py

```diff
@@ -38,6 +38,7 @@ def __init__(self, path: Union[str, Path]) -> None:
             self.path = path
 
         self.metadata = DatasetMeta.load(self._index_file)
+        self.table_cache: dict[str, tables.Table] = {}
```

Comment:
The other option is to do the caching in `sdg.py`, e.g. only for this particular step. I think it's better for it to be the default behavior in `Dataset`, especially since `dataset[table]` looks so harmless haha. But if there is a memory concern here with other steps I'm not aware of, let me know

Reply:
We could also introduce a flag later on to disable the caching if it becomes a problem

Reply:
Caching is a very nice solution, but I'd be a bit careful doing it implicitly for all datasets. One concern is memory as you say (e.g. backporting could be in a lot of trouble as it goes through basically all datasets) and the other is that it is a bit unexpected and could lead to side effects and hard to debug issues (can't think of any right now, but I can see myself struggling in the future :)). Moreover, I'm not sure if we would benefit from caching elsewhere.

If possible, I'd do caching for `sdg.py` exclusively and leave a note that reading feather is an expensive operation and can be cached.

By the way you can also do caching with [functools.cache](https://docs.python.org/3/library/functools.html) decorator.